lib/HLSL/DxilValidation.cpp

```diff
@@ -2134,17 +2134,38 @@ static void ValidateBarrierFlagArg(ValidationContext &ValCtx, CallInst *CI,
   }
 }
 
+std::string GetLaunchTypeStr(DXIL::NodeLaunchType LT) {
+  switch (LT) {
+	case DXIL::NodeLaunchType::Broadcasting:
+		return "Broadcasting";
+  case DXIL::NodeLaunchType::Coalescing:
+		return "Coalescing";
+  case DXIL::NodeLaunchType::Thread:
+    return "Thread";
+  case DXIL::NodeLaunchType::Invalid:
+  case DXIL::NodeLaunchType::LastEntry:
+    return "Invalid";
+
+  default:
+    return "Invalid";
+  }
+}
+
 static void ValidateDxilOperationCallInProfile(CallInst *CI,
                                                DXIL::OpCode opcode,
                                                const ShaderModel *pSM,
                                                ValidationContext &ValCtx) {
   DXIL::ShaderKind shaderKind =
       pSM ? pSM->GetKind() : DXIL::ShaderKind::Invalid;
   llvm::Function *F = CI->getParent()->getParent();
+  DXIL::NodeLaunchType nodeLaunchType = DXIL::NodeLaunchType::Invalid;
   if (DXIL::ShaderKind::Library == shaderKind) {
-    if (ValCtx.DxilMod.HasDxilFunctionProps(F))
+    if (ValCtx.DxilMod.HasDxilFunctionProps(F)) {
+      DxilModule &DM = ValCtx.DxilMod;
+      DxilEntryProps &entryProps = DM.GetDxilEntryProps(F);
+      nodeLaunchType = entryProps.props.Node.LaunchType;
       shaderKind = ValCtx.DxilMod.GetDxilFunctionProps(F).shaderKind;
```

Comment:
Doesn't matter too much in practice, but it feels weirdly inconsistent to capture `ValCtx.DxilMod` in `DM` for calculating `entryProps` but then use `ValCtx.DxilMod` directly again when getting `shaderKind`